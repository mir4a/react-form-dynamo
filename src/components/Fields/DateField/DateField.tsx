import React, { memo } from 'react';
import BaseField from '../BaseField';
import './DateField.css';

interface Props {
  initialValue?: string;
  label: string;
  name?: string;
  onChange?: Function;
  className?: string;
}

export default memo(function DateField(props: Props) {
  return <BaseField {...props} type="date" />;
});
