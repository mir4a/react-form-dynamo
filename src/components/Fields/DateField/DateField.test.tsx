import React from 'react';
import renderer from 'react-test-renderer';
import DateField from './DateField';

it('renders correctly', () => {
  const tree = renderer
    .create(<DateField label="A" onChange={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
