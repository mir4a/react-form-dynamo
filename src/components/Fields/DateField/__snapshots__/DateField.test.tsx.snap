// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <label
    className="Label Label__block"
  >
    <span
      className="Label-text"
    >
      A
    </span>
  </label>,
  <input
    className="Input DateField"
    onChange={[Function]}
    type="date"
    value=""
  />,
]
`;
